src/components/window/window.tsx

```diff
@@ -0,0 +1,74 @@
+import { FC, ReactNode } from 'react';
+import cn from 'classnames';
+
+import WindowHeader from './window-header';
+import WindowFooter from './window-footer';
+
+import '../../styles/components/window.scss';
+
+type WindowProps = {
+  width?: string;
+  height?: string;
+  title?: string;
+  withHeaderCloseButton?: boolean;
+  withFooterCloseButton?: boolean;
+  onWindowOpen?: () => void;
+  onWindowClose?: () => void;
+  withShadow?: boolean;
+  className?: string;
+  actionButtons?: ReactNode;
+  children?: ReactNode;
+};
+
+const Window: FC<WindowProps> = ({
+  width = '50vw',
+  height = '50vh',
+  title,
+  withHeaderCloseButton = false,
+  withFooterCloseButton = false,
+  onWindowOpen,
+  onWindowClose,
+  actionButtons,
+  withShadow = false,
+  className,
+  children,
+}) => {
+  const styles = {
+    width,
+    minHeight: height,
+    top: `calc((80vh - ${height}) / 2)`,
+    left: `calc((100vw - ${width}) / 2)`,
+  };
+  const baseClassName = 'window';
+  const cssClasses = cn(baseClassName, className, {
+    [`${baseClassName}--with-shadow`]: !!withShadow,
+  });
+  if (typeof onWindowOpen === 'function') {
```

Comment:
Could be replaced with `onWindowOpen?.()`

Reply:
Updated